ruuvi_examples/ruuvi_firmware/main.c

```diff
@@ -282,97 +296,149 @@ ret_code_t lis2dh12_int2_handler(const ruuvi_standard_message_t message)
 }
 
 
-/**
- * @brief Function for application main entry.
- */
+/**  This is where it all starts ++++++++++++++++++++++++++++++++++++++++++ 
+ main is entered as a result of one of SEVERAL events:
+  - Normal startup from press of reset button.
+  - Battery inserted.
+  - After DFU (Device Firmware Upgrade) at manufacturing Quality Assurance or user DFU.
+  - WatchDogTimer expiration and its interrupt handler didn't feed new value.
+  - Some error occured and
+  - Spontenous unknown reset.
+ All subsystems are initalized and any failures are noted and available later in init_status
+ Since some events occur after tag is deployed and no one can see the LEDs the system continues operating.
+
+ After initalizition (including setting up interrupts)
+    we loop here calling app_sched_execute and sd_app_evt_wait 
+*/
 int main(void)
 {
-  ret_code_t err_code = NRF_SUCCESS; // Error flag, OR any errors to it and check the error code at the end.
+ init_leds();                      // LEDs first (they're easy and cannot fail)  drivers/init/init.c ( RED got set up earlier)
+ nrf_delay_ms(1000u);              // sit here for a second with LEDs off
+
+  // Announce version
+ for ( int16_t i=0; i<3; i++){   // 3 segment version

```

Comment:
Reading blink codes is rather straining for anyone, especially factory tester. Version information is already available via NFC, whereas end-user without NFC reader can just refresh the firmware over the air or return the tag under warranty.

This also adds some seconds to test process which should be avoided if possible.

Reply:
This version does not blink actual error code, rather just an indication i.e 13 blinks.

Just pressing reset and reading the version blinks without any other equipment is just too easy not to be included.

Reply:
Is NFC readable via iPhone? 
How to read it from an android device?

May be  bluetooth_application_config.h might be differently named and moved to ruuvi_examples/ruuvi_firmware?

Reply:
Removed blinking of version (being outvoted).
 I could see this delay being a minor issue when producing and testing hundreds of tags
 I still believe that being able to see version without other utilities is preferable. 
Need to find NFC util for iPhone although apple is not cooperative in the NFC area. 
Power usage is not a concern as this (should) rarely occur.

Reply:
May be bluetooth_application_config.h might be differently named and moved to ruuvi_examples/ruuvi_firmware?
Why INIT_xWREV rather than simply VERSION ?
Is SW: a standard?

Reply:
iPhone 7 required for any iPhone NFC application! That's a too bad for me!